qml/utils.js

```diff
@@ -12,6 +12,12 @@ function versionGE(version, major, minor, micro) {
     return false
 }
 
+// Shim for String.prototype.startsWith(), added in Qt 5.8
+// Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith
+function startsWith(string, search, pos) {
+    return string.substr(!pos || pos < 0 ? 0 : +pos, search.length) === search
```

Comment:
How about `Math.max(0, pos || 0)` for the first argument?

Reply:
On second thought, when would you want to call this with `pos != 0`?

Reply:
Support for position is in actual ES6 API, I'm fine with dropping it in our version if you prefer.

Reply:
Ah, ok. I don't mind keeping it in.